src/main/java/uk/gov/hmcts/reform/draftstore/service/mappers/FromDbModelMapper.java

```diff
@@ -2,18 +2,30 @@
 
 import uk.gov.hmcts.reform.draftstore.domain.Draft;
 import uk.gov.hmcts.reform.draftstore.service.crypto.CryptoService;
+import uk.gov.hmcts.reform.draftstore.service.secrets.Secrets;
 
 public class FromDbModelMapper {
 
     public static Draft fromDb(
         uk.gov.hmcts.reform.draftstore.data.model.Draft dbDraft,
-        String secret
+        Secrets secrets
     ) {
+        String documentToReturn;
+
         // only during transition stage
-        String documentToReturn =
-            dbDraft.encryptedDocument == null
-                ? dbDraft.document
-                : CryptoService.decrypt(dbDraft.encryptedDocument, secret);
+        if (dbDraft.encryptedDocument == null) {
+            documentToReturn = dbDraft.document;
+        } else {
+            try {
+                documentToReturn = CryptoService.decrypt(dbDraft.encryptedDocument, secrets.primary);
+            } catch (Exception exc) {
+                if (secrets.secondary != null) {
+                    documentToReturn = CryptoService.decrypt(dbDraft.encryptedDocument, secrets.secondary);
+                } else {
+                    throw exc;
```

Comment:
Just to consider - the exception thrown here may be hard to understand, so we could wrap it in our exception, saying "Failed to decrypt the document from the draft".

Reply:
Oh, that could complicate converting exceptions to responses. Maybe not worth it, then.